docs/arguments.md

```diff
@@ -177,6 +177,32 @@ Flags for configuring KV store based on memberlist library. This feature is expe
    Timeout for writing 'packet' data.
 - `memberlist.transport-debug`
    Log debug transport messages. Note: global log.level must be at debug level as well.
+   
+#### Multi KV
+
+This is a special key-value implementation that uses two other KV stores (eg. consul, etcd or memberlist). One of them is always marked as primary, and all reads and writes go to primary store. Other one, secondary, is only used for writes. The idea is that operator can use multi KV store to migrate from primary to secondary store in runtime. For example, migration from Consul to Memberlist would look like this:
+
+- Set `ring.store` to use `multi` store. Set `-multi.primary=consul` and `-multi.secondary=memberlist`. All consul and memberlist settings must still be specified.
+- Start all Cortex microservices. They will still use Consul as primary KV, but they will also write share ring via memberlist.
+- Operator can now use "runtime config" mechanism to switch primary store to memberlist.
+- After all Cortex microservices have picked up new primary store, and everything looks correct, operator can now shut down Consul, and modify Cortex configuration to use `-ring.store=memberlist` only.
```

Comment:
Why don't we generally refer to the key-store instead of naming consul/etcd? It may look misleading that it only apply to migrating from consul to memberlist/etcd, while you can migrate from any store to any other store, right?

Reply:
I like to be specific in the example, and it also starts with "For example, migration from Consul to memberlist would look like this:". I'll modify it to use Consul -> Etcd as an example.